### PR TITLE
Change Tags

### PR DESCRIPTION
**Pull request by hodzy**
changes to functionality:
    1 - change of side bar to show only either (Student*in or Schuler*in)
    2 - allow showing of multiple tags (student positions) under the verwaltung tab
    3 - show praktikant in the Verwaltung tab when conditions are met

### CODE REVIEW

File: src/utils/UserUtils.ts [hunk 1]
```diff
@@ -4,20 +4,28 @@ export const getUserType = (user: User) => {
...
+
+export const getUserTags = (user: User) => {
+  const studentTags = {};
+  if (user.type === 'pupil') {
+    return { pupil: 'Schüler*in' };
+  }
 
   if (user.isTutor) {
-    return `Student*in`;
+    studentTags["isTutor"] = 'Tutor';
```
Comment (lsc0498) [inline src/utils/UserUtils.ts:17]: We should write `Tutor*in` instead of Tutor here :)

File: src/utils/UserUtils.ts [hunk 2]
```diff
@@ -4,20 +4,28 @@ export const getUserType = (user: User) => {
...
   }
 
   if (user.isInstructor) {
-    return 'Tutor*in';
+    studentTags["isInstructor"] =
+      user.university === null ? 'Kursleiter' : 'Praktikant';
```
Comment (lsc0498) [inline src/utils/UserUtils.ts:22]: Here as well `Kursleiter*in` and `Praktikant*in`

**Review by lsc0498 — APPROVED** [review]
Looks good to me. Thank you for your work!🎉

File: src/utils/UserUtils.ts [hunk 3]
```diff
@@ -4,20 +4,28 @@ export const getUserType = (user: User) => {
...
   }
 
   if (user.isProjectCoach) {
-    return 'Project-Coach';
+    studentTags["isProjectCoach"] = 'Project-Coach';
```
Comment (geroembser) [inline src/utils/UserUtils.ts:26]: And to be fully consistent:  it's `Projektcoach*in` here as well

Comment (geroembser) [reply src/utils/UserUtils.ts:26]: Additionally, typing `studentTags` and writing `studentTags.isProjectCoach = ` would be preferred and would fix eslint. This also applies for the other names above.

File: src/components/cards/SettingsCard.tsx [hunk 4]
```diff
@@ -112,13 +114,15 @@ const SettingsCard: React.FC<Props> = ({ user }) => {
...
           </div>
           <div className={classes.tagContainer}>
-            <Tag
-              background="#4E555C"
-              color="#ffffff"
-              style={{ marginLeft: '10px' }}
-            >
-              {getUserType(user)}
-            </Tag>
+            {Object.values(userTags).map((tag) => (
```
Comment (geroembser) [inline src/components/cards/SettingsCard.tsx:117]: What's the purpose of `userTags` being an object/dictionary while only using them as an array?

File: src/utils/UserUtils.ts (hunk 2 above)
Comment (geroembser) [inline src/utils/UserUtils.ts:22]: I don't think that only having university info suffices to be an intern (aka "Praktikant\*in"). What do you think @lsc0498? I'm not sure the current backend API is providing that information. And also due to the triple equal, we need to notice that `user.university` being undefined also results in being a "Praktikant\*in".

Comment (lsc0498) [reply src/utils/UserUtils.ts:22]: Actually I assumed that this would be a sufficient condition. Currently we have no other possibility to distinguish between `Praktikant*in` and `Kursleiter*in` given the existing backend api.
Concerning the triple equal, you're right we should use `user.university ?  'Praktikant' : 'Kursleiter'` instead.

Comment (geroembser) [reply src/utils/UserUtils.ts:22]: Ok. So if a university student registers for both being 1-on-1 tutor and course instructor (and not selecting the intern-flag and thus not providing the required information for interns), following those conditions here, she will be labelled as intern. Obviously this is wrong. 

But long story short: it looks like the backend has to provide the intern information - otherwise the frontend has no way to get this information here. So leaving it as proposed, there will be many people whose status will be displayed inappropriately.

Comment (hodzy) [reply src/utils/UserUtils.ts:22]: yes will change them

File: src/components/cards/SettingsCard.tsx (hunk 4 above)
Comment (hodzy) [reply src/components/cards/SettingsCard.tsx:117]: so the thing is, when we tried to search, if it is possible to return an array, it was shown that it is not possible, and so we used an object for only that reason
reference: "https://stackoverflow.com/questions/12059478/return-array-from-function"

File: src/utils/UserUtils.ts (hunk 2 above)
Comment (hodzy) [reply src/utils/UserUtils.ts:22]: @geroembser and  @lsc0498 you're right about the university, we shouls have made more general, but we tested for the value none, bec. we found out that it was always set to none when it was empty, and never was it set to undefined, but i guess the more general the better

Comment (hodzy) [reply src/utils/UserUtils.ts:22]: @geroembser so would you that we revert back to the previous way, we used to handle things (not showing prakitikant), until we provide new attributes on the backend side

File: src/utils/UserUtils.ts (hunk 3 above)
Comment (hodzy) [reply src/utils/UserUtils.ts]: so gero i can't understand exactly what you mean with that,
but generally we tried to idetify studentTags as new Object(), and hence could use the dot operator like that "studentTags.isProjectCoach = " and the code worked just fine, but then we couldn't really push it for some reason, if this error is relevent, i can try to replicate again, and take a screen shot of it.

File: src/utils/UserUtils.ts (hunk 2 above)
Comment (lsc0498) [reply src/utils/UserUtils.ts]: If there is some doubt about whether the university field is sufficient to determine whether a Student is `Praktikant*in` or not I would suggest to write `Kursleiter*in` in either case for now. At least it wouldn't be totally wrong I guess 😉😅

**Review by lsc0498 — COMMENTED** [review]
Now the build failed since e.g. `isTutor` is no member of `studentTags`. I would suggest to use a string array and append the corresponding strings using `push`.

**Comment by hodzy** [comment]
@lsc0498 
1- i have changed the type to array, although for some reason, it is still considered to be an object,and i couldn't find a fix for it 
2- the praktikant part is deleted out of the code

**Comment by lsc0498** [comment]
Looks good to me now👍 @geroembser are you fine with it too?

File: src/components/cards/SettingsCard.tsx (hunk 4 above)
Comment (lsc0498) [reply src/components/cards/SettingsCard.tsx:117]: Since `userTags` are an array now you can directly map from that array and omit this `Object.values` construction.

Comment (hodzy) [reply src/components/cards/SettingsCard.tsx:117]: @lsc0498 still for some reason, it's not working, so what i mean is, it is still considered to be an object, and when i try to cast it to an array which is compatible, and use the map function it concatenates all of them, and show all the tags in one tag

Comment (hodzy) [reply src/components/cards/SettingsCard.tsx:117]: and also shouldi merge the branch??

Comment (lsc0498) [reply src/components/cards/SettingsCard.tsx:117]: Just use `userTags.map`. That should actually work.

Comment (hodzy) [reply src/components/cards/SettingsCard.tsx]: change made

**Review by lsc0498 — APPROVED** [review]
Looks good now 👍 Thank you!

**Review by geroembser — APPROVED** [review]
Looks good! Thanks a lot!
